src/compose.ts

```diff
@@ -314,3 +332,64 @@ function normalizeKeyValuePairs(obj?: ListOrDict, sep: string = '='): Dict<strin
 		.fromPairs()
 		.value();
 }
+
+type Interpolator = (str: string) => string;
+
+function makeInterpolator(registry: Dict<string>): Interpolator {
+	return (str: string): string => {
+		return interpolate(registry, str);
+	};
+}
+
+function interpolate(registry: Dict<string>, str: string): string {
+	// see the tests for what sort of strings this regex matches against
+	const regex = /(\$\$?)(?:(\w+)|{(\w+)(?:([:\-\?]+)([^}]+))?})/g;
+	return str.replace(regex, (match, ...groups: Array<string | undefined>) => {
+		const [ escaped, name1, name2, operator, fallback ] = groups.slice(0, -2);
```

Comment:
Any reason this can't be simplified to:
```suggestion
		const [ escaped, name1, name2, operator, fallback ] = groups;
```
?